### ISSUE TITLE
`additionalPageIds` not respected for record update

### ISSUE THREAD

**Issue by hirnschmalz** [TYPO3-Solr/ext-solr#1655]
# Environment
* TYPO3 CMS: 7.6.22
* Solr version: 6.1.3
* PHP version: 5.6

# Expected behavior
If a record is changed it should be added to the queue.
# Actual behavior
Records outside of the siteroot are not re-indexed because in `processRecord` of `ApacheSolrForTypo3\Solr\IndexQueue\RecordMonitor` it is checked if the current record is configured in the siteroot of the record.
# Steps how to reproduce the issue
Create the following page structure
```
Root
  |-host1.domain.tld
    |- Sysfolder
  |-host2.domain.tld
```

If you now create a solr configuration for `host2.domain.tld` (add `additionalPageIds` to the queue configuration)  for records inside `Sysfolder` the index of `hsot2.domain.tld` will not be updated if records inside `Sysfolder` of `host1.domain.tld` are changed

**Comment by timohund** [comment]
@hirnschmalz This looks like the following scenario:

https://github.com/TYPO3-Solr/ext-solr/blob/master/Tests/Integration/IndexQueue/Fixtures/update_record_outside_siteroot.xml

It would be nice, when you could analyze in your system, what's wrong. 

Just to be sure:

```
useConfigurationTrackRecordsOutsideSiteroot
```

is set to 1 right?

**Comment by hirnschmalz** [comment]
Yes the `useConfigurationTrackRecordsOutsideSiteroot` extension setting value is set to `1`

**Comment by hirnschmalz** [comment]
Both sites contain solr sites

**Comment by timohund** [comment]
This could be the "reason" currently it needs to be "outside" the siteroot. It would be nice when you could investigate it. Otherwise we could offer you to do this, based on our support conditions.

**Comment by timohund** [comment]
@hirnschmalz What happens when you move the records outside both site roots and add them on both sites?

**Comment by dkd-kaehm** [comment]
### Fixed on EXT:solr 11.5.x+ in !!![TASK] Improve and Fix TSFE Initialization https://github.com/TYPO3-Solr/ext-solr/commit/a246cb8e34d80da6227445c0c800620f16264a00
